### PR TITLE
Lint client and data

### LINKED ISSUE #2578: lint_client.sh should probably prettify/lint/prune data/src as well

### PR DESCRIPTION
**Pull request by DrAnax**
Closes #2578 

Kept build_client.sh for now to avoid breaking actions.

### CODE REVIEW

**Comment by DarthGandalf** [comment]
the actions are configured in .github/workflows/
you can update them at the same time, without any breakage

**Comment by DrAnax** [comment]
Great, thanks

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -56,7 +56,7 @@ jobs:
         run: bash ./packages/client/set_version.sh
 
       - name: Lint the TypeScript code
-        run: bash ./packages/client/lint_client.sh
```
Comment (DarthGandalf) [inline .github/workflows/ci.yml:59]: Remove the old script?

File: packages/lint_scripts.sh [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#!/bin/bash
+
+set -e # Exit on any errors
+
+SECONDS=0
+
+# Get the directory of this script
+# https://stackoverflow.com/questions/59895/getting-the-source-directory-of-a-bash-script-from-within
+DIR_CLIENT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )/client"
+DIR_DATA="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )/data"
```
Comment (Zamiell) [inline packages/lint_scripts.sh:10]: doesn't satisfy DRY, i will fix it

File: .github/workflows/ci.yml (hunk 1 above)
Comment (DrAnax) [reply .github/workflows/ci.yml:59]: Forgot it, done

**Comment by DrAnax** [comment]
Linter should stop complaining once #2579 is accepted.

**Comment by Zamiell** [comment]
thank you, this was needed

**Pull request merged**